app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt

```diff
@@ -1103,6 +1106,51 @@ class AuthRepositoryImpl(
             }
     }
 
+    override suspend fun validatePIN(pin: String): ValidatePINResult {
+        val activeAccount = authDiskSource
+            .userState
+            ?.activeAccount
+            ?.profile
+            ?: return ValidatePINResult.Error
+        val privateKey = authDiskSource
+            .getPrivateKey(userId = activeAccount.userId)
+            ?: return ValidatePINResult.Error
+        val pinProtectedUserKey = authDiskSource
+            .getPinProtectedUserKey(userId = activeAccount.userId)
+            ?: return ValidatePINResult.Error
+
+        // HACK: As the SDK doesn't provide a way to directly validate the pin yet, we instead
```

Comment:
Modeled after the iOS code for consistency.

https://github.com/bitwarden/ios/blob/main/BitwardenShared/Core/Auth/Repositories/AuthRepository.swift#L782-L809

Reply:
SDK team has completed adding the function needed to validate pin without unlocking first. See https://bitwarden.atlassian.net/browse/PM-9527.

I suggest we update the SDK and test it in another PR after this one is merged.